bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -41,7 +43,9 @@ private SearchDecoration() {
 	 *                         the validation.
 	 */
 	public static boolean validateRegex(String regex, ControlDecoration targetDecoration) {
-		String errorMessage = getValidationError(regex);
+		GC gc = new GC(targetDecoration.getControl());
```

Comment:
Don't create the GC here. Create it in the method were it is actually needed (and dispose it there too).

Reply:
done